src/Admin/Editor.php

```diff
@@ -50,19 +50,28 @@ public function validate_and_pre_save_cb( $data, $post ) {
 		$document = new Document();
 
 		try {
-			if ( array_key_exists( 'post_content', $post ) && 'publish' === $post['post_status'] ) {
-				$data['post_content'] = $document->valid_or_throw( $post['post_content'], $post['ID'] );
+			// Check for empty post_content when publishing the query and throw
+			if ( 'publish' === $post['post_status'] && empty( $post['post_content'] ) ) {
+				throw new RequestError( __( 'Query string is empty', 'wp-graphql-smart-cache' ) );
 			}
+
+			$data['post_content'] = $document->valid_or_throw( $post['post_content'], $post['ID'] );
+
 		} catch ( RequestError $e ) {
 			AdminErrors::add_message( $e->getMessage() );
 
-			// Overwrite new/invalid query with previous working query, or empty
-			$existing_post = get_post( $post['ID'] );
-			if ( $existing_post && property_exists( $existing_post, 'post_content' ) ) {
-				try {
-					$data['post_content'] = $document->valid_or_throw( $existing_post->post_content, $post['ID'] );
-				} catch ( RequestError $e ) {
-					$data['post_content'] = '';
+			// If encountered invalid data when publishing query, revert some data. If draft, allow invalid query.
+			if ( 'publish' === $post['post_status'] ) {
+
+				// If has an existing published post and trying to publish with errors, bail before save_post
+				$existing_post = get_post( $post['ID'], ARRAY_A );
+				if ( $existing_post && 'publish' === $existing_post['post_status'] ) {
+
+					wp_safe_redirect( admin_url( sprintf( '/post.php?post=%d&action=edit', $post['ID'] ) ) );
+					exit;
+
+				} else {
+					$data['post_status'] = 'draft';
```

Comment:
```suggestion
					$data['post_status'] = 'draft';
					
					// This prevents the Admin UI from showing that the post has previously been published (because it actually hasn't been)
					$data['post_date_gmt'] = '0000-00-00 00:00:00';
```

Without this change, the admin UI shows that the document was published (even though it wasn't)

![CleanShot 2023-08-15 at 09 28 33](https://github.com/wp-graphql/wp-graphql-smart-cache/assets/1260765/88de163a-59b2-491a-a051-61a4847cd205)

With this change, the admin UI no longer falsely shows the publish date: 

![CleanShot 2023-08-15 at 09 30 39](https://github.com/wp-graphql/wp-graphql-smart-cache/assets/1260765/99ac1277-c719-4127-830d-362e0a52b667)

Reply:
If a date is already in post_date, it will be overwritten with your suggestion.  I think I'll check $existing_post and use that if exists.  For post_date and post_data_gmt.